### PR TITLE
[INLONG-5302][Manager] Supports the extension of data consumption for different MQs

### LINKED ISSUE #5302: [Feature][Manager] Supports the extension of data consumption for different MQs
### Description

Supports the extension of data consumption for different MQs.

For example, only the consumption configuration of TubeMQ and Pulsar is currently supported.
After the refactor, it supports the rapid expansion of other types of MQ, such as Kafka, RocketMQ, and so on.

### Design

1. The table DDL:
```sql
-- ----------------------------
-- Table structure for inlong_consume
-- ----------------------------
CREATE TABLE IF NOT EXISTS `inlong_consume`
(
    `id`               int(11)      NOT NULL AUTO_INCREMENT COMMENT 'Incremental primary key',
    `consumer_group`   varchar(256) NOT NULL COMMENT 'Consumer group name, filled in by the user, undeleted ones cannot be repeated',
    `description`      varchar(256)          DEFAULT '' COMMENT 'Inlong consume description',
    `mq_type`          varchar(10)           DEFAULT 'TUBEMQ' COMMENT 'Message queue type, high throughput: TUBEMQ, high consistency: PULSAR',
    `topic`            varchar(256) NOT NULL COMMENT 'The target topic of this consume',
    `inlong_group_id`  varchar(256) NOT NULL COMMENT 'The target inlong group id of this consume',
    `filter_enabled`   int(2)                DEFAULT '0' COMMENT 'Whether to filter consume, 0: not filter, 1: filter',
    `inlong_stream_id` varchar(256)          DEFAULT NULL COMMENT 'The target inlong stream id of this consume, needed if the filter_enabled=1',
    `ext_params`       text                  DEFAULT NULL COMMENT 'Extended params, will be saved as JSON string',
    `in_charges`       varchar(512) NOT NULL COMMENT 'Name of responsible person, separated by commas',
    `status`           int(4)                DEFAULT '100' COMMENT 'Inlong consume status',
    `is_deleted`       int(11)               DEFAULT '0' COMMENT 'Whether to delete, 0: not deleted, > 0: deleted',
    `creator`          varchar(64)  NOT NULL COMMENT 'Creator name',
    `modifier`         varchar(64)           DEFAULT NULL COMMENT 'Modifier name',
    `create_time`      timestamp    NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT 'Create time',
    `modify_time`      timestamp    NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP COMMENT 'Modify time',
    `version`          int(11)      NOT NULL DEFAULT '1' COMMENT 'Version number, which will be incremented by 1 after modification',
    PRIMARY KEY (`id`),
    UNIQUE KEY `unique_inlong_consume` (`consumer_group`, `is_deleted`)
) ENGINE = InnoDB
  DEFAULT CHARSET = utf8mb4 COMMENT ='Inlong consume table';
```

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ciscozhou**
### Prepare a Pull Request

- Fixes #5302 

### Motivation

*Supports the extension of data consumption for different MQs*

### Modifications

1. Merge table consumption and consumption_pulsar into table inlong_consume.
2. Build the code of inlong_consume according to the consumption.

### Verifying this change
(Please pick either of the following options)

- [ ] This change is a trivial rework/code cleanup without any test coverage.
 - [ ] This change is already covered by existing tests, such as:
(please describe tests)
 - [ ] This change added tests and can be verified as follows:
 
### Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
- If a feature is not applicable for documentation, explain why?
- If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

**Comment by healchow** [comment]
Thanks for your contribution🎉

Please add more info for the PR.

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/consume/InlongConsumeServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    public boolean isConsumerGroupExists(String consumerGroup, Integer excludeSelfId) {
+        ConsumptionQuery consumptionQuery = new ConsumptionQuery();
+        consumptionQuery.setConsumerGroup(consumerGroup);
+        consumptionQuery.setIsAdminRole(true);
+        List<InlongConsumeEntity> result = consumeEntityMapper.listByQuery(consumptionQuery);
+        if (excludeSelfId != null) {
+            result = result.stream().filter(consumer -> !excludeSelfId.equals(consumer.getId()))
+                    .collect(Collectors.toList());
+        }
+        return !CollectionUtils.isEmpty(result);
```
Comment (leosanqing) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/consume/InlongConsumeServiceImpl.java:97]: Could use apache.commons.collectionUtils.CollectionUtils.isNotEmpty()
Easier to understand

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/consume/InlongConsumeOperatorFactory.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+@Service
+public class InlongConsumeOperatorFactory {
+
+    @Autowired
+    private List<InlongConsumeOperator> consumeOperatorList;
+
+    /**
+     * Get a consumption operator instance via the given mqType
+     */
+    public InlongConsumeOperator getInstance(String mqType) {
```
Comment (leosanqing) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/consume/InlongConsumeOperatorFactory.java:37]:
```suggestion
    public InlongConsumeOperator getInstance(String mqType) {
        return consumeOperatorList.stream()
                .filter(inst -> inst.accept(mqType))
                .findFirst()
                .orElseThrow(() -> new BusinessException(String.format(ErrorCodeEnum.MQ_TYPE_NOT_SUPPORTED.getMessage(), mqType)));
    }
```

File: inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/consumption/pulsar/ConsumePulsarDTO.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+@AllArgsConstructor
+@ApiModel("Inlong group info for Pulsar")
+public class ConsumePulsarDTO {
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper(); // thread safe
+
+    @ApiModelProperty(value = "Consumption ID")
+    private Integer consumptionId;
+
+//    @ApiModelProperty(value = "Consumer group")
```
Comment (leosanqing) [inline inlong-manager/manager-pojo/src/main/java/org/apache/inlong/manager/pojo/consumption/pulsar/ConsumePulsarDTO.java:48]: Personal suggestion, new classes, it is best not to keep comments, delete them directly,

Same as below

File: inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongConsumeController.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    private InlongGroupService groupService;
+    @Autowired
+    private InlongGroupProcessService groupProcessOperation;
+    @Autowired
+    private InlongConsumeService consumeService;
+
+    @RequestMapping(value = "/consume/save", method = RequestMethod.POST)
+    @OperationLog(operation = OperationType.CREATE)
+    @ApiOperation(value = "Save inlong consume")
+    public Response<Integer> save(@Validated @RequestBody InlongConsumeRequest consumeRequest) {
```
Comment (leosanqing) [inline inlong-manager/manager-web/src/main/java/org/apache/inlong/manager/web/controller/InlongConsumeController.java:65]: Create and update different field checks, such as Id, you can refer to this class

org.apache.inlong.manager.web.controller.StreamTransformController#update

File: inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/InlongConsumeEntity.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    private Integer id;
+    private String inlongGroupId;
+    private String consumerGroup;
+    private Integer filterEnabled;
+    private String inlongStreamId;
+//    private String name;
+    private String description;
+    private String mqType;
+    private String topic;
+//    private String mqResource;
```
Comment (healchow) [inline inlong-manager/manager-dao/src/main/java/org/apache/inlong/manager/dao/entity/InlongConsumeEntity.java:41]: Just remove those fields if they were not necessary.

File: inlong-manager/manager-dao/src/main/resources/mappers/InlongConsumeEntityMapper.xml [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+        <result column="is_deleted" jdbcType="INTEGER" property="isDeleted"/>
+        <result column="creator" jdbcType="VARCHAR" property="creator"/>
+        <result column="modifier" jdbcType="VARCHAR" property="modifier"/>
+        <result column="create_time" jdbcType="TIMESTAMP" property="createTime"/>
+        <result column="modify_time" jdbcType="TIMESTAMP" property="modifyTime"/>
+        <result column="version" jdbcType="INTEGER" property="version"/>
+    </resultMap>
+
+    <sql id="Base_Column_List">
+        id
```
Comment (woofyzhao) [inline inlong-manager/manager-dao/src/main/resources/mappers/InlongConsumeEntityMapper.xml:44]: strange line feed

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/consume/InlongConsumeServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+        LOGGER.debug("begin to save inlong consume={} by user={}", request, operator);
+        Preconditions.checkNotNull(request, "inlong consume request cannot be null");
+        Preconditions.checkNotNull(request.getTopic(), "inlong consume topic cannot be null");
+        String consumerGroup = request.getConsumerGroup();
+        Preconditions.checkNotNull(consumerGroup, "inlong consume topic cannot be null");
+        if (consumerGroupExists(consumerGroup, request.getId())) {
+            throw new BusinessException(String.format("consumer group %s already exist", consumerGroup));
+        }
+
+        //if (GroupStatus.notAllowedUpdate(curStatus)) {
```
Comment (woofyzhao) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/consume/InlongConsumeServiceImpl.java:78]: Should the commented code be removed?

Comment (ciscozhou) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/consume/InlongConsumeServiceImpl.java]: removed

**Review by healchow — APPROVED** [review]
LGTM

**Review by woofyzhao — APPROVED** [review]

**Review by leosanqing — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**